hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -896,6 +907,40 @@ public void reclaimPreviouslyReservedThrottle(int n, HederaFunctionality functio
         synchronizedThrottleAccumulator.leakUnusedThrottlePreviouslyReserved(n, function);
     }
 
+    @Override
+    public boolean shouldThrottleNOfUnscaled(int n, HederaFunctionality function) {
+        return synchronizedThrottleAccumulator.shouldThrottleNOfUnscaled(n, function, userTransactionConsensusTime);
+    }
+
+    public boolean shouldThrottleTxn(TransactionInfo txInfo) {
+        return networkUtilizationManager.shouldThrottle(txInfo, current());
+    }
+
+    @Override
+    public boolean hasThrottleCapacityForChildTransactions() {
+        var isAllowed = true;
+        final var childRecords = recordListBuilder.childRecordBuilders();
+
+        for (int i = 0, n = childRecords.size(); i < n && isAllowed; i++) {
+            final var childRecord = childRecords.get(i);
+            if (Objects.equals(childRecord.status(), SUCCESS)) {
+                final var tx = childRecord.build().transaction();
+                final var txInfo = new TransactionInfo(
+                        tx, childRecord.transactionBody(), tx.sigMap(), tx.signedTransactionBytes(), functionality);
+
+                if (txInfo.functionality() == CONTRACT_CREATE || txInfo.functionality() == CONTRACT_CALL) {
+                    continue;
+                }
+
+                if (shouldThrottleTxn(txInfo)) {
+                    isAllowed = false;
+                }
+            }
+        }
+
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Here we are missing the reseting of the throttles from snapshots in the case of which throttling occurs i.e. [this from mono](https://github.com/hashgraph/hedera-services/blob/5a68acf8bdda2a8656104490d774bfa4afd20223/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/records/TxnAwareRecordsHistorian.java#L197). Is that on purpose?

Reply:
Definitely not on purpose, it is updated